ui/src/components/Canvas.tsx

```diff
@@ -1344,12 +1394,16 @@ export function Canvas() {
       checkNodesEndLocation(event, [currentNode], "ROOT");
       //clear the pasting state
       setPasting(null);
+      // delete the original (hidden) node
+      if (cutting) {
+        reactFlowInstance.deleteElements({ nodes: [{ id: cutting }] });
```

Comment:
When cutting, once the user left-click to drop the node, it will trigger:
1. add the new node (the `addPod(apolloClient, ...)` above)
2. delete the original hidden node.

I'm having some trouble with 2. This line seems to trigger `deletePod(null, ...)` in `nodes.tsx`, so there won't be a remote `deletePod` graphql query. But I do see a `deletePod` query. Can you point out where does that `deletePod` query come from?

Reply:
I see. It should come from `onNodesDelete`.

Reply:
Yes, it triggers onNodeChanges's `delete` type.

I generalize the logic in https://github.com/codepod-io/codepod/pull/184. This PR is almost done except I need a better implementation to create a scope as well as its children in DB at once. You can try that PR to see if it is the expected behavior of scope cut-copy-paste.

Reply:
Ah, I see, thanks. Yeah, that PR is a great reference on this. I'll take a close look.